CHANGELOG.md

```diff
@@ -1,5 +1,99 @@
 ### [Versions](https://material-ui.com/versions/)
 
+## 4.12.0
+###### _Nov 14 2020_
+
+Big thanks to the 10 contributors who made this release possible.
+
+This release is intended to help and prepare the migration to Material-UI v5 ⏫:
+
+- 📚 Add deprecations to anticipate v5. A deprecation is added every time a breaking change in v5 allows it. If no warnings are reported in the console, you are set. Each warning comes with a simple message that explains how to handle the deprecation.
```

Comment:
> A deprecation is added every time a breaking change in v5 allows it. If no warnings are reported in the console, you are set.

Unfortunately this isn't true. There's a long list of breaking changes in v5 that don't yet have a deprecation in v4. If someone reads this and thinks they are set for v5, they're in for a world of pain.

I have pinged the contributors of those breaking changes that don't have a deprecation on the respective PR (only once per contributor - some have multiple PRs.) I didn't for those created by @mui-org/core-team as I assume they'll take care of them in due course.

Reply:
"allows it" could be emphasized.

Reply:
That doesn't magically make it true.

Reply:
? The message is: when possible we do it, when not, we don't.

Reply:
> A deprecation is added every time a breaking change in v5 allows it.

A deprecation _should_ be added every time a breaking change in v5 allows, but that isn't the case. Like I said™️: There's a long list of breaking changes in v5 that don't yet have a deprecation in v4. 

I even went on to explain what I'd done to try and help address that, which should have made it abundantly clear what the problem is with the current wording. A full stop doesn't mean stop reading. Do let me know if you read this last sentence.

Reply:
> There's a long list of breaking changes in v5 that don't yet have a deprecation in v4.

You are considering not v4 latest vs. v5 latest, I don't think that it matters. I think that it should target v4 latest vs. v5 stable.

Reply:
I'm confused, I don't understand.

Reply:
> A deprecation is added every time a breaking change in v5 allows it

"is added" as was already added or will be before we release v5.

Reply:
```suggestion
- 📚 Start to add deprecations to anticipate v5. A deprecation is added every time a breaking change in v5 **allows it**. Each warning comes with a simple message that explains how to handle the deprecation. If no warnings are reported in the console, you are set for this first batch.
```

Reply:
We can't add deprecation warnings for every breaking change. Deprecation warnings are just there to help you identify where you have to do something in your codebase. 

For example, if you rely on some classes via CSS that are now removed, we can't add a warning if we recognize them.